src/Task/Assets/ServiceProvider.php

```diff
@@ -0,0 +1,22 @@
+<?php
+namespace Robo\Task\Assets;
+
+use League\Container\ServiceProvider\AbstractServiceProvider;
+
+class ServiceProvider extends AbstractServiceProvider
+{
+    protected $provides = [
+        'taskMinify',
+        'taskImageMinify',
+        'taskLess',
+        'taskScss',
+    ];
+
+    public function register()
+    {
+        $this->getContainer()->add('taskMinify', Minify::class);
+        $this->getContainer()->add('taskImageMinify', ImageMinify::class);
+        $this->getContainer()->add('taskLess', Less::class);
+        $this->getContainer()->add('taskScss', Scss::class);
+    }
```

Comment:
I have an idea to simplify that declaration.
Let's create an abstract class

```php
class TaskProvider extends AbstractServiceProvider 
{
protected $tasks = [];

function provides($service)
{
     if (array_key_exists($service, $this->tasks)) {
         return $this->getContainer()->get($service);
     }
}

//
function register() 
{
     foreach ($this->tasks as $name => $class) {
         $this->getContainer()->add($name, $class);
     }
}
```

if we do that change, provider class changes just to:

```php
class AssetTaskProvider extends TaskProvider
protected $tasks = [
    'taskMinify' => Minify::class,
    // ....
];
```